common/reflect_resource_plugin_framework.go

```diff
@@ -289,3 +290,105 @@ func checkTheStringInForceSendFields(fieldName string, forceSendFields []string)
 	}
 	return false
 }
+
+func pluginFrameworkTypeToSchema(v reflect.Value) map[string]schema.Attribute {
```

Comment:
This would also need to include aliases and tracking context that we currently have to typeToSchema for sdkv2? 

Example -- we use tracking context for depth:
````
		if tc.depthExceeded(typeField) {
			// Skip the field if recursion depth is over the limit.
			log.Printf("[TRACE] over recursion limit, skipping field: %s, max depth: %d", getNameForType(typeField.Type), tc.getMaxDepthForTypeField(typeField))
			continue
		}
````

and aliases while getting field name:

`````
fieldName := chooseFieldNameWithAliases(typeField, v.Type(), aliases)
`````

Reply:
For tracking context yes, alias no.

Tracking context is coming up as the next step for recursion depth limit.

In the future all of alias information will go directly into the proto layer with annotations, so we don't have to worry about them in tf